ledger-service/http-json-testing/src/main/scala/com/daml/http/HttpServiceUserFixture.scala

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.model.headers.Authorization
+import akka.stream.Materializer
+import com.daml.grpc.adapter.{AkkaExecutionSequencerPool, ExecutionSequencerFactory}
+import com.daml.http.HttpServiceTestFixture.{authorizationHeader, postRequest}
+import com.daml.http.util.ClientUtil.uniqueId
+import com.daml.jwt.JwtSigner
+import com.daml.jwt.domain.{Jwt, DecodedJwt}
+import com.daml.platform.sandbox.SandboxRequiringAuthorizationFuns
+import scalaz.syntax.show._
+import scalaz.syntax.tag._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait HttpServiceUserFixture {
+  protected def testId: String
+
+  // TODO(SC) we make asys, mat, and aesf "win" to match the prior behavior when
+  // these defns were in AbstractHttpServiceIntegrationTestFuns.  But it
+  // might be possible to phase them out entirely in favor of AkkaBeforeAndAfterAll
+  import shapeless.tag, tag.@@ // used for subtyping to make `AHS ec` beat executionContext
+  implicit val `AHS asys`: ActorSystem @@ this.type = tag[this.type] {
```

Comment:
What is the use of tagging these implicits?

Reply:
Would prefer a solution that does not need tagging.

Reply:
I'ld also like to know why we use tagging here. No blocker to me though.

Reply:
I've included this as e73865f586930d8c0393dc15b00c10e3275a1f2f. NB: it reverts #11109 so we may revisit in case of increased CI timeouts.

Shapeless tags make strict subtypes of the left-hand side of `@@`; the implicit resolver will prefer an implicit with a stricter subtype. When you can't use the "priority" trick on companions, you can make things subtypes instead. `@@ this.type` is a pretty strict subtype.

#13115 will include a more self-documenting way to introduce subtypes for this purpose, tentatively called `with ImplicitPreference`, but switching to that definitely ought to be separate from this PR.

Attempting to remove the high-priority `AHS ec` (introduced in #3936 5b52999da2858) resulted in 4af98e1d27efddb2465f030e71e5b941a2144b4f, so it doesn't seem worth attempting to remove.

Reply:
Thanks for the feedback. We can improve this further in a gardening ticket.  
A simple solution would likely be to just define
`implicit val ec = system.dispatcher`
In the tests itself where the ec is needed, instead of in a fixture trait, something simple like that would have my preference.